api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java

```diff
@@ -51,11 +54,36 @@ public void bufferBillingProject() {
     billingProjectBufferEntryDao.save(entry);
   }
 
+  public void syncBillingProjectStatus() {
+    BillingProjectBufferEntry entry = billingProjectBufferEntryDao
+        .findFirstByStatusOrderByCreationTimeAsc(StorageEnums.billingProjectBufferStatusToStorage(CREATING));
+
+    if (entry == null) {
+      return;
+    }
+
+    switch (fireCloudService.getBillingProjectStatus(entry.getFireCloudProjectName()).getCreationStatus()) {
+      case READY:
+        entry.setStatusEnum(AVAILABLE);
```

Comment:
opt: could be nice to log the time it took to create the project. Terra should track this as well, but it's been a long time coming